operator/api/vectorized/v1alpha1/cluster_types.go

```diff
@@ -1418,3 +1408,155 @@ func (r *Cluster) GetDecommissionBrokerID() *int32 {
 func (r *Cluster) SetDecommissionBrokerID(id *int32) {
 	r.Status.DecommissioningNode = id
 }
+
+// getNodePoolsFromSpec returns the NodePools defined in the spec.
+// This contains the primary NodePool (driven by spec.replicas for example), and the
```

Comment:
nit: does "primary" refers to the "default" / "legacy" node pool here? If yes, maybe we should use "default" or "legacy" instead

Reply:
fixed